packages/hash/api/src/collab/EntityWatcher.ts

```diff
@@ -51,11 +59,35 @@ export class EntityWatcher {
      * @todo address this
      */
     if (msg.action !== "D") {
-      const entityVersion = EntityVersion.parseWal2JsonMsg(msg);
+      let message: RealtimeMessage;
+      switch (msg.table) {
+        case "entity_versions":
+          message = {
+            table: msg.table,
+            record: EntityVersion.parseWal2JsonMsg(msg),
+          };
+          break;
+        case "link_versions":
+          message = {
+            table: msg.table,
+            record: LinkVersion.parseWal2JsonMsg(msg),
+          };
+          break;
+        case "aggregation_versions":
+          message = {
+            table: msg.table,
+            record: AggregationVersion.parseWal2JsonMsg(msg),
+          };
+          break;
+        default:
+          throw new Error(
+            `Unexpected record from table '${msg.table}' when processing message.`,
+          );
+      }
```

Comment:
This works totally fine, so doesn't need to be changed. But if I'm understanding correctly, the type-guard on line 51 will prevent the default case of this switch from occurring so this could become:
```suggestion
      const message: RealtimeMessage =
        table === "entity_versions"
          ? { table, record: EntityVersion.parseWal2JsonMsg(msg) }
          : table === "link_versions"
          ? { table, record: LinkVersion.parseWal2JsonMsg(msg) }
          : { table, record: AggregationVersion.parseWal2JsonMsg(msg) };
```

This would have the added benefit that if a developer adds a new table name to the `supportedRealtimeTables` array, this line will immediately throw a TS error if the new table name isn't handled.

(If there is a use-case for supporting a new table, and not having to handle it here, then please ignore this comment)